lib/msf/core/web_services/servlet/event_servlet.rb

```diff
@@ -22,9 +22,9 @@ def self.get_event
       warden.authenticate!
       begin
         sanitized_params = sanitize_params(params, env['rack.request.query_hash'])
-        data = get_db.events(sanitized_params)
+        data, pagination_total = get_db.events(sanitized_params)
         data = data.first if is_single_object?(data, sanitized_params)
-        set_json_data_response(response: data)
+        set_json_data_response(response: data, merge_data: { 'pagination_total' => pagination_total })
```

Comment:
Just confirming, does `merge_data` get used in the `set_json_data_response` method?

https://github.com/rapid7/metasploit-framework/blob/92be1cc74ac360ed61bf58b84380c21f2683dc2c/lib/msf/core/web_services/servlet_helper.rb#L27-L30

Reply:
I'd be inclined to revert the `set_json_response` back to its previous implementation, and update the calling code to handle the pagination instead

```ruby
data, pagination_total = get_db.events(sanitized_params)
data = data.first if is_single_object?(data, sanitized_params)
data = data.merge({ 'pagination_total' => pagination_total })

get_json_data_response(response: data)
```

Reply:
No, data is an array type, it cannot merge hash types, and this will destroy the original data structure